packages/cli/commands/theme/marketplace-validate.js

```diff
@@ -35,67 +31,99 @@ exports.handler = async options => {
   await loadAndValidateOptions(options);
 
   const accountId = getAccountId(options);
-  const absoluteSrcPath = path.resolve(getCwd(), src);
-  let stats;
+
+  trackCommandUsage('validate', null, accountId);
+
+  const spinnies = new Spinnies();
+
+  spinnies.add('marketplaceValidation', {
+    text: i18n(`${i18nKey}.logs.validatingTheme`, {
+      path: src,
+    }),
+  });
+
+  // Kick off validation
+  let requestResult;
+  const assetType = 'THEME';
+  const requestGroup = 'EXTERNAL_DEVELOPER';
   try {
-    stats = fs.statSync(absoluteSrcPath);
-    if (!stats.isDirectory()) {
-      logger.error(
-        i18n(`${i18nKey}.errors.invalidPath`, {
-          path: src,
-        })
-      );
-      return;
-    }
-  } catch (e) {
-    logger.error(
-      i18n(`${i18nKey}.errors.invalidPath`, {
-        path: src,
-      })
-    );
-    return;
+    requestResult = await requestValidation(accountId, {
+      path: src,
+      assetType,
+      requestGroup,
+    });
+  } catch (err) {
+    logger.debug(err);
   }
 
-  if (!options.json) {
-    logger.log(
-      i18n(`${i18nKey}.logs.validatingTheme`, {
-        path: src,
-      })
-    );
+  // Poll till validation is finished
+  try {
+    const checkValidationStatus = async () => {
+      let validationStatus;
+      if (requestResult) {
+        validationStatus = await getValidationStatus(accountId, {
+          validationId: requestResult,
+        });
+      }
+
+      if (validationStatus && validationStatus === 'REQUESTED') {
+        await new Promise(resolve => setTimeout(resolve, 2000));
```

Comment:
`2000` would be good as a constant at the top of the file, like `SLEEP_TIME` or something